cparse.nim

```diff
@@ -714,7 +714,7 @@ proc structPragmas(p: Parser, name: PNode, origName: string): PNode =
   #addSon(pragmas, newIdentNodeP("pure", p), newIdentNodeP("final", p))
   if p.options.useHeader:
     addSon(pragmas,
-      newIdentStrLitPair(p.options.importcLit, p.currentNamespace & origName, p),
+      newIdentStrLitPair(p.options.importcLit, "struct " & p.currentNamespace & origName, p),
```

Comment:
This should only be done if we are not in CPP mode.

Reply:
https://github.com/nim-lang/Nim/issues/7900 fails on CPP mode if I remove this patch. CPP mode works fine with both `struct` and `class`, i.e. if I tested it correctly.

Reply:
It's ugly, C++ doesn't have these silly "struct" prefixes.